pkg/domain/infra/abi/quadlet.go

```diff
@@ -0,0 +1,748 @@
+			}
+			return filepath.SkipDir
+		}
+		info, err := d.Info()
+		if err != nil {
+			if !errors.Is(err, fs.ErrNotExist) {
+				logrus.Warnf("Error descending into path %s: %v", path, err)
+			}
+			return filepath.SkipDir
+		}
+		if !info.IsDir() {
+			name := info.Name()
+			if strings.HasPrefix(name, ".") && strings.HasSuffix(name, ".app") {
+				err := readLinesFromFile(path, func(line string) error {
+					reverseMap[line] = name
+					appMap[name] = append(appMap[name], line)
+					return nil
+				})
+				if err != nil {
+					return err
+				}
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, nil, err
+	}
+	return reverseMap, appMap, nil
+}
+
+// Get the paths of all quadlets available to the current user
+func getAllQuadletPaths() []string {
+	var quadletPaths []string
+	quadletDirs := systemdquadlet.GetUnitDirs(rootless.IsRootless())
+	for _, dir := range quadletDirs {
+		dents, err := os.ReadDir(dir)
+		if err != nil {
+			if !errors.Is(err, fs.ErrNotExist) {
+				// This is perfectly normal, some quadlet directories aren't created by the package
+				logrus.Warnf("Cannot list Quadlet directory %s: %v", dir, err)
+			}
+			continue
+		}
+		logrus.Debugf("Checking for quadlets in %q", dir)
+		for _, dent := range dents {
+			if systemdquadlet.IsExtSupported(dent.Name()) && !dent.IsDir() {
+				logrus.Debugf("Found quadlet %q", dent.Name())
+				quadletPaths = append(quadletPaths, filepath.Join(dir, dent.Name()))
+			}
+		}
+	}
+	return quadletPaths
+}
+
+// Generate systemd service name for a Quadlet from full path to the Quadlet file
+func getQuadletServiceName(quadletPath string) (string, error) {
+	unit, err := parser.ParseUnitFile(quadletPath)
+	if err != nil {
+		return "", fmt.Errorf("parsing Quadlet file %s: %w", quadletPath, err)
+	}
+
+	serviceName, err := systemdquadlet.GetUnitServiceName(unit)
+	if err != nil {
+		return "", fmt.Errorf("generating service name for Quadlet %s: %w", filepath.Base(quadletPath), err)
+	}
+	return serviceName + ".service", nil
+}
+
+type QuadletFilter func(q *entities.ListQuadlet) bool
+
+func generateQuadletFilter(filter string, filterValues []string) (func(q *entities.ListQuadlet) bool, error) {
+	switch filter {
+	case "name":
+		return func(q *entities.ListQuadlet) bool {
+			res := util.StringMatchRegexSlice(q.Name, filterValues)
+			return res
+		}, nil
+	default:
+		return nil, fmt.Errorf("%s is not a valid filter", filter)
+	}
+}
+
+func (ic *ContainerEngine) QuadletList(ctx context.Context, options entities.QuadletListOptions) ([]*entities.ListQuadlet, error) {
+	reverseMap, _, err := buildAppMap(systemdquadlet.GetInstallUnitDirPath(rootless.IsRootless()))
+	if err != nil {
+		return nil, fmt.Errorf("unable to build app map: %w", err)
+	}
+	// Is systemd available to the current user?
+	// We cannot proceed if not.
+	conn, err := dbus.NewWithContext(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("connecting to systemd dbus: %w", err)
+	}
```

Comment:
That is why the list test is broken, you are running against the root socket all the time, this must use `conn, err := systemd.ConnectToDBUS()` like you did in install and remove.